File: src/main/java/com/rabbit/api/dto/request/UpdateCompanyRequest.java
package com.rabbit.api.dto.request;

import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.MultipartFile;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateCompanyRequest {

    private String description;

    private String addressLine1;

    private String addressLine2;

    private String city;

    private String state;

    private String country;

    @Pattern(regexp = "^[0-9]{6}$", message = "PIN code must be 6 digits")
    private String pinCode;

    @Pattern(regexp = "^[0-9]{10}$", message = "Contact number must be 10 digits")
    private String contactNumber;

    private MultipartFile logo;
}

File: dtos/requests/UpdateCompanyRequest.js
class UpdateCompanyRequest {
  constructor({ description, addressLine1, addressLine2, city, state, country, pinCode, contactNumber }) {
    this.description = description || null;
    this.addressLine1 = addressLine1 || null;
    this.addressLine2 = addressLine2 || null;
    this.city = city || null;
    this.state = state || null;
    this.country = country || null;
    this.pinCode = pinCode || null;
    this.contactNumber = contactNumber || null;
  }
}
module.exports = UpdateCompanyRequest;
